.github/workflows/ci-testing.yml

```
@@ -42,7 +42,9 @@ jobs:
           pip list
 
       - name: Tests
-        run: coverage run --source litdata -m pytest tests -v --durations=100
+        run: |
+          pip install pytest-xdist
+          coverage run --source litdata -m pytest tests -v --durations=100 -n auto
```

```suggestion
          coverage run --source litdata -m pytest tests -v --durations=100 -n 2
```
lets start just with 2 and we can bump the scale later